workers/ohsome_quality_analyst/indicators/mapping_saturation/models.py

```diff
@@ -0,0 +1,408 @@
+"""Statistical Model Classes"""
+
+import os
+from abc import ABC, abstractmethod
+
+import numpy as np
+import rpy2.robjects.packages as rpackages
+from rpy2 import robjects
+from scipy.optimize import curve_fit
+from scipy.stats.distributions import t as t_distribution
+
+from ohsome_quality_analyst.indicators.mapping_saturation import metrics
+
+
+class BaseStatModel(ABC):
+    """Base Statistical Model"""
+
+    def __init__(self, xdata: np.ndarray, ydata: np.ndarray):
+        assert np.shape(xdata) == np.shape(ydata)
+        self.xdata = xdata
+        self.ydata = ydata
+        self.coefficients = {}
+        self.fitted_values = None
+        self.asym_conf_int = None
+
+    @property
+    @abstractmethod
+    def name(self) -> str:
+        """Name of the statistical model"""
+        pass
+
+    @property
+    @abstractmethod
+    def function_formula(self) -> str:
+        """Function formular of the statistical model"""
+        pass
+
+    @property
+    @abstractmethod
+    def asymptote(self) -> str:
+        pass
+
+    @property
+    def mae(self) -> str:
+        """Mean absolute error"""
+        return metrics.mae(self.ydata, self.fitted_values)
+
+    def as_dict(self):
+        return {
+            key: getattr(self, key)
+            for key in (
+                "name",
+                "function_formula",
+                "asymptote",
+                "mae",
+                "xdata",
+                "ydata",
+                "coefficients",
+                "fitted_values",
+            )
+        }
+
+
+class Sigmoid(BaseStatModel):
```

Comment:
I'm not sure if this model always returns what we expect. E.g. for Athens the plot looks like this. Are you sure that the asymptote is correct? From the plot I would expect that it could be higher, but maybe I'm wrong.

![image](https://user-images.githubusercontent.com/7045979/146796388-9953fe82-bae1-4107-8962-0429ae28d36e.png)

Reply:
For context here are the plots of all the models which produced fits:
![Figure_11](https://user-images.githubusercontent.com/83658582/146909660-e0ace721-7b7a-4d79-8576-ed14b44a2385.png)

Reply:
From the above figure the plot with the smallest Mean Absolute Error is in the middle.
Using the latest commit this is also the plot which will be choosen and displayed on the website.
![screenshort1](https://user-images.githubusercontent.com/83658582/146910048-3bb80125-a401-4e95-9f71-aa47a2dc8859.png)

Reply:
I think your plot does not refer to the same regions a my plot. My plot was for Athens.

Reply:
I just double checked using the website and the script for creating the plots and the above plots do refer to region of Athen (for me it is fid=1)